datahub-web-react/src/app/entity/shared/components/styled/search/action/DomainsDropdown.tsx

```diff
@@ -1,27 +1,78 @@
-import React from 'react';
+import { message, Modal } from 'antd';
+import React, { useState } from 'react';
+import { useBatchSetDomainMutation } from '../../../../../../../graphql/mutations.generated';
+import { SetDomainModal } from '../../../../containers/profile/sidebar/Domain/SetDomainModal';
 import ActionDropdown from './ActionDropdown';
 
 type Props = {
     urns: Array<string>;
     disabled: boolean;
+    refetch?: () => void;
 };
 
 // eslint-disable-next-line
-export default function DomainsDropdown({ urns, disabled = false }: Props) {
-    return (
-        <ActionDropdown
-            name="Domains"
-            actions={[
-                {
-                    title: 'Set domain',
-                    onClick: () => null,
-                },
-                {
-                    title: 'Unset domain',
-                    onClick: () => null,
+export default function DomainsDropdown({ urns, disabled = false, refetch }: Props) {
+    const [isEditModalVisible, setIsEditModalVisible] = useState(false);
+    const [batchSetDomainMutation] = useBatchSetDomainMutation();
+
+    const batchUnsetDomains = async () => {
+        try {
+            await batchSetDomainMutation({
+                variables: {
+                    input: {
+                        resources: [...urns.map((urn) => ({ resourceUrn: urn }))],
+                    },
                 },
-            ]}
-            disabled={disabled}
-        />
+            });
+            message.success({ content: 'Removed Domain!', duration: 2 });
```

Comment:
not that it matters much and I know this is how it's been in your other batch PRs, but is there a particular reason we are using an `async` function and awaiting the result, as opposed to using the promise`.then(() => {} ).catch(() => {})` way that we do most other places?

Reply:
No we can change it to keep consistency, I think this is copied from another place we use

Reply:
updating!